jena-arq/src/main/java/org/apache/jena/sparql/core/mem/QuadTableForm.java

```diff
@@ -124,31 +96,21 @@ public PMapQuadTable get() {
 			return new PMapQuadTable(name()) {
 
 				@Override
-				protected Quad quad(final Node s, final Node p, final Node o, final Node g) {
-					return Quad.create(g, s, p, o);
-				}
+                public Quad tuple(final Quad q) {
+                    return create(q.getSubject(), q.getPredicate(), q.getObject(), q.getGraph());
```

Comment:
It'll be clear when the white space is standardised but am I right in seeing:

* Using Quad and putting strange things in the slots which is very confusing. `getGraph` returns the `getSubject`.
* There seems to be some creating objects just to take them apart again. The strength of the design is that quads/triples are not stored, instead nodes are placed in indexes. To reflect that, operations based on g/s/p/o in some order are clearer to me at least as well as reducing object churn.